codalab/worker/rest_client.py

```diff
@@ -135,32 +134,18 @@ def _upload_with_chunked_encoding(
         else:
             conn = http.client.HTTPSConnection(parsed_base_url.netloc, timeout=TIMEOUT)
         with closing(conn):
-            conn.putrequest(method, parsed_base_url.path + path)
-
-            # Set headers.
             headers = {
                 'Authorization': 'Bearer ' + self._get_access_token(),
-                'Transfer-Encoding': 'chunked',
                 'X-Requested-With': 'XMLHttpRequest',
             }
-            headers.update(self._extra_headers)
-            for header_name, header_value in headers.items():
-                conn.putheader(header_name, header_value)
-            conn.endheaders()
-
-            # Use chunked transfer encoding to send the data through.
-            bytes_uploaded = 0
-            while True:
-                to_send = fileobj.read(CHUNK_SIZE)
-                if not to_send:
-                    break
-                conn.send(b'%X\r\n%s\r\n' % (len(to_send), to_send))
-                bytes_uploaded += len(to_send)
-                if progress_callback is not None:
-                    should_resume = progress_callback(bytes_uploaded)
```

Comment:
But now we're not invoking `progress_callback`, which is important...

Reply:
true - I assumed this was a tradeoff we were making for simplicity. If we need `progress_callback`, I do not think we can use python's chunk encoding. @epicfaace thoughts?

Reply:
Actually, yeah, `progress_callback` is important and we need to keep it. I guess that means we can't use Python's chunk encoding then.

Reply:
Closing pr